File: 01-Arrays/20250730-intermediate/go/solution_test.go
package main

import (
	"reflect"
	"testing"
)

func TestMergeSortedArrays(t *testing.T) {
	tests := []struct {
		name     string
		arr1     []int
		arr2     []int
		expected []int
	}{
		{
			name:     "Basic merge",
			arr1:     []int{1, 3, 5},
			arr2:     []int{2, 4, 6},
			expected: []int{1, 2, 3, 4, 5, 6},
		},
		{
			name:     "Empty first array",
			arr1:     []int{},
			arr2:     []int{1, 2, 3},
			expected: []int{1, 2, 3},
		},
		{
			name:     "Empty second array",
			arr1:     []int{1, 2, 3},
			arr2:     []int{},
			expected: []int{1, 2, 3},
		},
		{
			name:     "Both arrays empty",
			arr1:     []int{},
			arr2:     []int{},
			expected: []int{},
		},
		{
			name:     "Different lengths",
			arr1:     []int{1, 5, 9, 10, 15, 20},
			arr2:     []int{2, 3, 8, 13},
			expected: []int{1, 2, 3, 5, 8, 9, 10, 13, 15, 20},
		},
		{
			name:     "First array larger elements",
			arr1:     []int{4, 5, 6},
			arr2:     []int{1, 2, 3},
			expected: []int{1, 2, 3, 4, 5, 6},
		},
		{
			name:     "Duplicate elements",
			arr1:     []int{1, 3, 5},
			arr2:     []int{1, 3, 5},
			expected: []int{1, 1, 3, 3, 5, 5},
		},
		{
			name:     "Negative numbers",
			arr1:     []int{-5, -1, 2},
			arr2:     []int{-3, 0, 4},
			expected: []int{-5, -3, -1, 0, 2, 4},
		},
		{
			name:     "Single elements",
			arr1:     []int{5},
			arr2:     []int{3},
			expected: []int{3, 5},
		},
		{
			name:     "Large ranges",
			arr1:     []int{100, 200, 300},
			arr2:     []int{150, 250, 350, 400, 500},
			expected: []int{100, 150, 200, 250, 300, 350, 400, 500},
		},
	}

	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			result := MergeSortedArrays(test.arr1, test.arr2)
			if !reflect.DeepEqual(result, test.expected) {
				t.Errorf("MergeSortedArrays(%v, %v) = %v; expected %v", test.arr1, test.arr2, result, test.expected)
			}
		})
	}
}

File: 01-Arrays/20250730-intermediate/ts/solution.test.ts
import { mergeSortedArrays } from './solution';
import testCases from './test-cases.json';

describe('mergeSortedArrays', () => {
    test('Basic merge', () => {
        const result = mergeSortedArrays([1, 3, 5], [2, 4, 6]);
        expect(result).toEqual([1, 2, 3, 4, 5, 6]);
    });

    test('Empty first array', () => {
        const result = mergeSortedArrays([], [1, 2, 3]);
        expect(result).toEqual([1, 2, 3]);
    });

    test('Empty second array', () => {
        const result = mergeSortedArrays([1, 2, 3], []);
        expect(result).toEqual([1, 2, 3]);
    });

    test('Both arrays empty', () => {
        const result = mergeSortedArrays([], []);
        expect(result).toEqual([]);
    });

    test('Different lengths', () => {
        const result = mergeSortedArrays([1, 5, 9, 10, 15, 20], [2, 3, 8, 13]);
        expect(result).toEqual([1, 2, 3, 5, 8, 9, 10, 13, 15, 20]);
    });

    test('First array larger elements', () => {
        const result = mergeSortedArrays([4, 5, 6], [1, 2, 3]);
        expect(result).toEqual([1, 2, 3, 4, 5, 6]);
    });

    test('Duplicate elements', () => {
        const result = mergeSortedArrays([1, 3, 5], [1, 3, 5]);
        expect(result).toEqual([1, 1, 3, 3, 5, 5]);
    });

    test('Negative numbers', () => {
        const result = mergeSortedArrays([-5, -1, 2], [-3, 0, 4]);
        expect(result).toEqual([-5, -3, -1, 0, 2, 4]);
    });

    test('Single elements', () => {
        const result = mergeSortedArrays([5], [3]);
        expect(result).toEqual([3, 5]);
    });

    test('Large ranges', () => {
        const result = mergeSortedArrays([100, 200, 300], [150, 250, 350, 400, 500]);
        expect(result).toEqual([100, 150, 200, 250, 300, 350, 400, 500]);
    });
});